erpnext/selling/doctype/sales_order/sales_order.py

```diff
@@ -660,6 +661,15 @@ def validate_serial_no_based_delivery(self):
 					).format(item.item_code)
 				)
 
+	def validate_reserved_stock(self):
+		"""Clean reserved stock flag for not stocked Item"""
+
+		for item in self.items:
+			if not cint(
+				frappe.get_cached_value("Item", item.item_code, "is_stock_item")
+			) or not frappe.db.get_single_value("Stock Settings", "enable_stock_reservation"):
+				item.reserve_stock = 0
```

Comment:
get `enable_stock_reservation` from DB once and use it in the iteration.

Why we are fetching `is_stock_item` from Item master as you have added a field is_stock_item in the Sales Order Item itself?